### PR TITLE
Sticky Footer

### PR DESCRIPTION
**Pull request by ChazUK**
Adding in a flex based sticky footer.

### CODE REVIEW

**Comment by ChazUK** [comment]
[Sticky Footer](https://trello.com/c/CcKC3B11/585-sticky-footer)

File: src/assets/scss/modules/_sticky-footer.scss [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  margin: 0;
+  padding: 0;
+}
+
+app-root {
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+
+  > * {
```
Comment (omarCreativeDev) [inline src/assets/scss/modules/_sticky-footer.scss]: why was this needed?

File: src/app/app.component.html [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <app-header></app-header>
 
-<div class="govuk-width-container">
+<div id="main" class="govuk-width-container">
```
Comment (omarCreativeDev) [inline src/app/app.component.html]: could we use the <main> tag instead?

File: src/assets/scss/modules/_sticky-footer.scss (hunk 1 above)
Comment (ChazUK) [reply src/assets/scss/modules/_sticky-footer.scss]: For some reason the width would reduce to like 50%, I'm guessing it's something to do with flex turning child elements to inline-block

File: src/app/app.component.html (hunk 2 above)
Comment (ChazUK) [reply src/app/app.component.html]: This template has been copied from https://design-system.service.gov.uk/styles/page-template/custom/index.html, and we need flex working on app header, footer and the div in between them. Annoyingly the `main` tag is too nested
